linter/src/crow/crow.cpp

```diff
@@ -24,9 +26,50 @@ get_server_thread()
             std::string uid = req.url_params.get("uid");
             std::string algo_id = req.url_params.get("algo_id");
 
+            // Watchdog to check for crash/pending after 130s
+            std::thread check_pending_thread([&algo_id = algo_id, &uid = uid]() {
+                std::this_thread::sleep_for(std::chrono::seconds(130));
+
+                // Check status from Firebase
+                nutc::client::LintingResultOption linting_status =
+                    nutc::client::get_algo_status(uid, algo_id);
+
+                // If status is pending, force push a failure
+                if (linting_status == nutc::client::LintingResultOption::PENDING) {
+                    // Push failure
+                    std::string error_msg =
+                        "unknown runtime error: your code is syntactically correct but "
+                        "timed out (130 seconds) while linting";
```

Comment:
```suggestion
                        "timed out after 130 seconds while linting";
```

also will need to make this time configurable